src/t_stream.c

```diff
@@ -1386,6 +1401,174 @@ robj *createObjectFromStreamID(streamID *id) {
                         id->ms,id->seq));
 }
 
+/* Returns non-zero if the ID is 0-0. */
+int streamIDEqZero(streamID *id) {
+    return !(id->ms || id->seq);
+}
+
+/* A helper that returns non-zero if the range from 'start' to the stream's
+ * end doesn't contain a tombstone.
+ *
+ * NOTE: this assumes that the caller had verified that 'start' is less than
+ * 's->last_id'. */
+int streamRangeDoesNotContainTombstone(stream *s, streamID *start) {
+    streamID start_id;
+
+    if (!s->length || streamIDEqZero(&s->max_deleted_entry_id)) {
+        /* The stream is empty or has no tombstones. */
+        return 1;
+    }
+
+    /* Copy start ID, if given, or default to 0-0. */
+    if (start) {
+        start_id = *start;
+    } else {
+        start_id.ms = 0;
+        start_id.seq = 0;
+    }
+
+    if (streamCompareID(&s->first_id,&s->max_deleted_entry_id) > 0) {
+        /* The latest tombstone is before the first entry. */
+        return 1;
+    }
+
+    if (streamCompareID(&start_id,&s->max_deleted_entry_id) >= 0) {
+        /* The range doesn't include a tombstone. */
+        return 1;
+    }
+
+    /* The range includes a tombstone. */
+    return 0;
+}
+
+/* Replies with a consumerg group's current lag, that is the number of messages
+ * in the stream that are yet to be delivered. In case that the lag isn't
+ * available due to fragmentation, the reply to the client is a null. */
+void streamReplyWithCGLag(client *c, stream *s, streamCG *cg) {
+    int valid = 1;
```

Comment:
please start with `valid = 0` (i know it's gonna take more lines, but it's gonna be clearer